test/integration/text_layer_spec.mjs

```diff
@@ -39,15 +39,21 @@ describe("Text layer", () => {
     }
 
     function middlePosition(rect) {
-      return { x: rect.x + rect.width / 2, y: rect.y + rect.height / 2 };
+      return {
```

Comment:
I checked out this patch locally, but unfortunately these changes aren't enough to fix #18774. I'd therefore suggest we remove the changes to this file from this PR and work on this test in a separate PR, to speed up landing this PR first because the highlight test is more important as it fails on the bots.

(I initially thought both issues shared the same root cause after playing around with rounding in `getSpanRect`, but I should probably have thought about that better before suggesting we try to fix that one too in this PR; sorry for that.)

Reply:
It's probably because I just keep the floor around the division because with a previous version, rect.x was an integer but now it isn't anymore.
That said it's probably better to do that in an other patch.